### PR TITLE
[dv/kmac] add lc_escalate_en sequence

### PR DESCRIPTION
**Pull request by cindychip**
This PR has two commits:
1). Add DV sequence to drive lc_escalete_en_i signal and check kmac goes to terminal state.
2). Update design assertions to handle lc_escalate_en_i.

### CODE REVIEW

**Review by weicaiyang — APPROVED** [review]
LGTM with a question

File: hw/ip/kmac/data/kmac_base_testplan.hjson [hunk 1]
```diff
@@ -133,6 +133,20 @@
       milestone: V2
       tests: ["{variant}_error"]
     }
+    {
+      name: lc_escalation
+      desc: '''
+            Randomly set `lc_escalate_en` to `On` during Kmac operations.
```
[the remark this answers is not in the document]
Comment (cindychip) [reply hw/ip/kmac/data/kmac_base_testplan.hjson:139]: For kmac, lc_escalate_en is everyone Off, unless `On` is asserted. Does that match with the rest of the IPs？

[the remark this answers is not in the document]
Comment (cindychip) [reply hw/ip/kmac/data/kmac_base_testplan.hjson:139]: Thanks for checking on that! Actually filed a design issue and updated the testbench for that.

**Review by eunchan — COMMENTED** [review]
LGTM design part. I will check lc comparison logic

**Comment by cindychip** [comment]
Thanks Eunchan! The issue is here: https://github.com/lowRISC/opentitan/issues/10828
Hope it is clear.

**Review by weicaiyang — APPROVED** [review]

**Pull request merged**